sdk/core/Azure.Core/src/Response.cs

```diff
@@ -144,5 +144,8 @@ internal static void DisposeStreamIfNotBuffered(ref Stream? stream)
                 stream = null;
             }
         }
+
+		// TODO: Are we happy with this internal property on Response?
```

Comment:
It's reasonable, but why would we not add a property pointing to the whole message, i.e. the classifier is already in the message.

Reply:
Classifier is the only thing that survives the message lifetime.

Reply:
My concern about adding the whole message is that it contains `Request` which may be disposed before the `Response` lifetime is over, so we would have to take extra care in `Response` to only access parts of message at the right time.  This approach side-steps that issue.

Reply:
But if we are careful we can reference the entire thing.

Reply:
> But if we are careful we can reference the entire thing.

Do you mean that we can include message as an internal property as long as we make sure we don't access `Request` after it is disposed?  Just out of curiosity, what is the benefit of having the whole message over just the `ResponseClassifier`?

Reply:
>Do you mean that we can include message as an internal property as long as we make sure we don't access Request after it is disposed? 

Yes

>Just out of curiosity, what is the benefit of having the whole message over just the ResponseClassifier?

Next time we need something from the message we don't need to add another property. But it's a pretty weak reason for me.